libcxx/include/__iterator/bounded_iter.h

```diff
@@ -67,7 +72,13 @@ struct __bounded_iter {
   _LIBCPP_HIDE_FROM_ABI __bounded_iter(__bounded_iter const&) = default;
   _LIBCPP_HIDE_FROM_ABI __bounded_iter(__bounded_iter&&)      = default;
 
-  template <class _OtherIterator, __enable_if_t< is_convertible<_OtherIterator, _Iterator>::value, int> = 0>
+  template < class _OtherIterator,
+             __enable_if_t<
+                 _And<is_constructible<_Iterator, const _OtherIterator&>,
+                      is_convertible<const _OtherIterator&, _Iterator>,
```

Comment:
Do you need to check both `is_convertible` and `is_constructible`? Doesn't convertibility subsume constructibility?

Reply:
I'll change back to use `is_convertible` only...

However, [allocator.requirements.general] (and perhaps many other places in the library specification) sometimes only says "(implicit) conversion"/"(implicitly) convertible", while C++ types can be so tricky that explicit and implicit conversions can have different semantics, or the explicit conversion can be ill-formed while the implicit one is not. Maybe an LWG issue, requiring conversions between fancy pointers to be sane, is desired.